### PR TITLE
do not add duplicate jars to MetaData containerResources

### PR DESCRIPTION
**Pull request by lachlan-roberts**

### CODE REVIEW

**Review by janbartel — DISMISSED** [review]
I'm ok with this change, however probably should output a WARN that there is a duplicate jar on the container path. Also do same for `ee9`.

**Comment by janbartel** [comment]
+1

**Review by janbartel — DISMISSED** [review]

**Review by olamy — DISMISSED** [review]

File: jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaData.java [hunk 1]
```diff
@@ -715,7 +715,10 @@ public List<Resource> getContainerResources()
     public void addContainerResource(Resource jar)
     {
         Objects.requireNonNull(jar);
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaData.java]: The null check is an incomplete check.
What if the `jar` isn't readable?
We have `Resources.isReadable*()` methods for this reason.

File: jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/MetaData.java [hunk 2]
```diff
@@ -714,7 +714,11 @@ public List<Resource> getContainerResources()
 
     public void addContainerResource(Resource jar)
     {
-        _orderedContainerResources.add(jar);
+        Objects.requireNonNull(jar);
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/MetaData.java]: Ditto, null check feels like a partial solution here.

**Review by joakime — APPROVED** [review]

**Comment by joakime** [comment]
@lachlan-roberts what project is this PR for?

**Pull request merged**